example/v2_framework.cpp

```diff
@@ -0,0 +1,418 @@
+
+static_assert(is_of_type<metre*(second* metre), derived_unit<power<struct metre, 2>, struct second>>);
+static_assert(is_of_type<second*(metre* metre), derived_unit<power<struct metre, 2>, struct second>>);
+
+static_assert(is_of_type<1 / second * metre, derived_unit<struct metre, per<struct second>>>);
+static_assert(is_of_type<1 / second * second, struct one>);
+
+static_assert(is_of_type<second / one, struct second>);
+static_assert(is_of_type<1 / second / one, derived_unit<struct one, per<struct second>>>);
+
+static_assert(is_of_type<metre / second * second, struct metre>);
+static_assert(is_of_type<1 / second * (1 / second), derived_unit<struct one, per<power<struct second, 2>>>>);
+static_assert(is_of_type<1 / (second * second), derived_unit<struct one, per<power<struct second, 2>>>>);
+static_assert(is_of_type<1 / (1 / (second * second)), derived_unit<power<struct second, 2>>>);
+
+static_assert(is_of_type<metre / second * (1 / second), derived_unit<struct metre, per<power<struct second, 2>>>>);
+static_assert(
+  is_of_type<metre / second*(metre / second), derived_unit<power<struct metre, 2>, per<power<struct second, 2>>>>);
+static_assert(is_of_type<metre / second*(second / metre), struct one>);
+
+static_assert(is_of_type<watt / joule, derived_unit<struct watt, per<struct joule>>>);
+static_assert(is_of_type<joule / watt, derived_unit<struct joule, per<struct watt>>>);
+
+// comparisons of equivalent dimensions
+// static_assert(metre / metre == one);
+
+// static_assert(1 / second == dim_frequency);
+// static_assert(1 / dim_frequency == second);
+// static_assert(dim_frequency * second == one);
+
+// static_assert(metre * metre == dim_area);
+// static_assert(metre * metre != dim_volume);
+// static_assert(dim_area / metre == metre);
+
+// static_assert(metre * metre * metre == dim_volume);
+// static_assert(dim_area * metre == dim_volume);
+// static_assert(dim_volume / metre == dim_area);
+// static_assert(dim_volume / metre / metre == metre);
+// static_assert(dim_area * dim_area / metre == dim_volume);
+// static_assert(dim_area * (dim_area / metre) == dim_volume);
+// static_assert(dim_volume / (metre * metre) == metre);
+
+// static_assert(metre / second == dim_speed);
+// static_assert(metre * second != dim_speed);
+// static_assert(metre / second / second != dim_speed);
+// static_assert(metre / dim_speed == second);
+// static_assert(dim_speed * second == metre);
+
+// static_assert(metre / second / second == dim_acceleration);
+// static_assert(metre / (second * second) == dim_acceleration);
+// static_assert(dim_speed / second == dim_acceleration);
+// static_assert(dim_speed / dim_acceleration == second);
+// static_assert(dim_acceleration * second == dim_speed);
+// static_assert(dim_acceleration * (second * second) == metre);
+// static_assert(dim_acceleration / dim_speed == dim_frequency);
+
+
+// Bq + Hz should not compile
+
+// Bq + Hz + 1/s should compile?
```

Comment:
Additive operators should be associative. If `Bq + Hz + 1/s` doesn't compile, `1/s + Bq + Hz` shouldn't either.

Reply:
Totally agree. This refers to: https://github.com/mpusz/units/discussions/389#discussioncomment-3594634.